apps/settings/js/modules/bluetooth/bluetooth_connection_manager.js

```diff
@@ -0,0 +1,674 @@
+            return this._connect(device).then(() => {
+            }, (reason) => {
+              Debug('connect(): ' + 
+                    'Cannot connect device, reject with reason: ' + reason);
+              return Promise.reject(reason);
+            });
+          }, (reason) => {
+            Debug('connect(): ' + 
+                  'Cannot disconnect current connected device, ' + 
+                  'reject with reason: ' + reason);
+            return Promise.reject(reason);
+          });
+        } else {
+          // Have checked there is no connected device in map now.
+          return this._connect(device).then(() => {
+          }, (reason) => {
+            Debug('connect(): ' + 
+                  'Cannot disconnect current connected device, ' + 
+                  'reject with reason: ' + reason);
+            return Promise.reject(reason);
+          });
+        }
+      }, (reason) => {
+        Debug('connect(): Cannot get connected device, ' + 
+              'reject with reason: ' + reason);
+        return Promise.reject(reason);
+      });
+    },
+
+    /**
+     * The method will connect the input device with the device's adapter.
+     *
+     * @access private
+     * @memberOf BluetoothConnectionManager
+     * @param {Object BluetoothDevice} device
+     * @returns {Promise}
+     */
+    _connect: function btcm__connect(device) {
+      if (!this._defaultAdapter) {
+        return Promise.reject('default adapter is not existed!!');
+      }
+
+      // Save the connecting address
+      this.connectingAddress = device.address;
+      Debug('_connect(): start connecting device = ' + device);
+
+      return this._defaultAdapter.connect(device).then(() => {
+        // At least one profile is connected.
+        // Clean up the connecting address
+        // this.connectingAddress = null;
+        // Save the connected address
+        // this.connectedAddress = device.address;
+        // var options = {
+        //   address: device.address,
+        //   connected: true
+        // };
+        // this._updateConnectionStatus(options);
+        Debug('_connect(): resolve, connectedAddress = ' + 
+              this.connectedAddress);
+      }, () => {
+        // No available profiles are connected.
+        this.connectingAddress = null;
+        Debug('_connect(): reject with connection failed');
+        return Promise.reject('connection failed');
+      });
+    },
+
+    /**
+     * The method will disconnect the input device with the device's adapter.
+     *
+     * @access public
+     * @memberOf BluetoothConnectionManager
+     * @param {Object BluetoothDevice} device
+     * @returns {Promise}
+     */
+    disconnect: function btcm_disconnect(device) {
+      if (!this._defaultAdapter) {
+        return Promise.reject('default adapter is not existed!!');
+      }
+
+      return this._defaultAdapter.disconnect(device).then(() => {
+        // this.connectedAddress = null;
```

Comment:
It seems `connectedAddress` is updated in `_updateConnectionStatus` right? If so, please remove these lines.